canister/src/state.rs

```diff
@@ -90,6 +90,22 @@ impl State {
     }
 }
 
+#[cfg(target_arch = "wasm32")]
+fn get_current_time_in_secs() -> u64 {
+    // to get seconds from nanoseconds
+    ic_cdk::api::time() / 1_000_000_000
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn get_current_time_in_secs() -> u64 {
+    use std::time::SystemTime;
+
+    SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs()
+}
```

Comment:
Is `state.rs` the right place to put this logic? I'd create a method called `fn time()` in `runtime.rs` that returns the time in ns. That's where we put logic that is different across runtimes.

Reply:
Done.